x-pack/plugins/lens/public/editor_frame_service/editor_frame/editor_frame.tsx

```diff
@@ -30,37 +32,45 @@ import {
   switchToSuggestion,
 } from './suggestion_helpers';
 import { trackUiEvent } from '../../lens_ui_telemetry';
+import {
+  useLensSelector,
+  useLensDispatch,
+  LensAppState,
+  DispatchSetState,
+  onChangeFromEditorFrame,
+} from '../../state_management';
 
 export interface EditorFrameProps {
-  doc?: Document;
   datasourceMap: Record<string, Datasource>;
   visualizationMap: Record<string, Visualization>;
-  initialDatasourceId: string | null;
-  initialVisualizationId: string | null;
   ExpressionRenderer: ReactExpressionRendererType;
   palettes: PaletteRegistry;
   onError: (e: { message: string }) => void;
   core: CoreStart;
   plugins: EditorFrameStartPlugins;
-  dateRange: {
-    fromDate: string;
-    toDate: string;
-  };
-  query: Query;
-  filters: Filter[];
-  savedQuery?: SavedQuery;
-  searchSessionId: string;
-  onChange: (arg: {
-    filterableIndexPatterns: string[];
-    doc: Document;
-    isSaveable: boolean;
-  }) => void;
   showNoDataPopover: () => void;
   initialContext?: VisualizeFieldContext;
 }
 
 export function EditorFrame(props: EditorFrameProps) {
-  const [state, dispatch] = useReducer(reducer, props, getInitialState);
+  const {
+    filters,
+    searchSessionId,
+    savedQuery,
+    query,
+    persistedDoc,
+    indexPatternsForTopNav,
+    lastKnownDoc,
```

Comment:
We can probably resolve this in a follow-up, but it seems like the dependence on `lastKnownDoc` in here triggers an additional render of the frame on state change:
* User changes something
* Editor frame state updates - frame renders
* Effect hook updates the top level redux state
* Editor frame renders again because `lastKnownDoc` got updated
* Editor frame renders again after active data gets updated

The `lastKnownDocRef` shielded the editor frame from re-rendering on step 4

Not a blocker for this PR

Reply:
I hope we'll remove this one completely once refactoring Editor frame parts so for now I'll leave it as is.